File: 2015/day10/day10.cpp
#include<iostream>
#include<string>
using namespace std;

string answerfunction(const string& lines) {
    string newString;
    int i = 0;
    int length = lines.length();

    while (i < length) {
        if (lines[i] != lines[i + 1]) {
            newString += '1';
            newString += lines[i];
            i++;
        } else {
            int count = 1;
            do {
                count += 1;
                i++;
            } while (lines[i] == lines[i + 1]);
            newString += to_string(count);
            newString += lines[i];
            i++;
        }
    }

    return newString;
}

int main() {
    string lines = "1321131112";
    for (int i = 0; i < 40; i++) {
        lines = answerfunction(lines);
    }
    int final = lines.length();
    std::cout << "Part One: "<< final << std::endl;
    string lines1 = "1321131112";
    for (int i = 0; i < 50; i++) {
        lines1 = answerfunction(lines1);
    }
    int final1 = lines1.length();
    std::cout << "Part Two: " << final1 << std::endl;
}

File: 2015/day10/day10.py
def answer_function(lines):
    new_string = ""
    i = 0
    length = len(lines)

    while i < length:
        if i + 1 < length and lines[i] != lines[i + 1]:
            new_string += '1' + lines[i]
            i += 1
        else:
            count = 1
            while i + 1 < length and lines[i] == lines[i + 1]:
                count += 1
                i += 1
            new_string += str(count) + lines[i]
            i += 1

    return new_string

def main():
    lines = "1321131112"
    for _ in range(40):
        lines = answer_function(lines)
    
    final = len(lines)
    print("Part One:", final)

    lines1 = "1321131112"
    for _ in range(50):
        lines1 = answer_function(lines1)

    final1 = len(lines1)
    print("Part Two:", final1)

if __name__ == "__main__":
    main()
